src/MSBuild.UnitTests/PerfLog_Tests.cs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading;
+
+using Microsoft.Build.CommandLine;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+using Microsoft.Build.UnitTests.Shared;
+using Xunit;
+using Xunit.Abstractions;
+using Shouldly;
+using System.IO.Compression;
+using System.Reflection;
+
+namespace Microsoft.Build.UnitTests
+{
+    public class PerfLogTests
+    {
+#if USE_MSBUILD_DLL_EXTN
+        private const string MSBuildExeName = "MSBuild.dll";
+#else
+        private const string MSBuildExeName = "MSBuild.exe";
+#endif
+
+        private readonly ITestOutputHelper _output;
+
+        public PerfLogTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact]
+        [Trait("Category", "mono-osx-failing")] // Disable on Mono OSX, since Mono doesn't implement EventSource.
+        public void TestPerfLogEnabledProducedLogFile()
+        {
+            // Setup perf log.
+            string perfLogDir = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString("N"));
```

Comment:
I think all this would be easier with a TestEnvironment.

Reply:
Can you please point me to an example that I should follow?

Reply:
I like [this example](https://github.com/dotnet/msbuild/blob/6945be012e19946fb89118f1c1b37c5ff33c5a9b/src/Build.UnitTests/Construction/SolutionProjectGenerator_Tests.cs#L89-L100), but I'm a little biased. You can also use `env.SetEnvironmentVariable(<name>, <value>);`.

Reply:
Using the test environment should make this simpler (since it handles cleanup), and allow avoiding the RunnerUtilities changes.

Reply:
Yes, this definitely cleaned things up.